### PR TITLE
no-issue: cache npm dependencies on CI

### PR DESCRIPTION
**Pull request by theghostwhoforks**
cache npm on CI for faster builds

### CODE REVIEW

File: .github/workflows/mvn_test.yml [hunk 1]
```diff
@@ -26,6 +26,17 @@ jobs:
             ${{ runner.os }}-build-${{ env.cache-name }}-
             ${{ runner.os }}-build-
             ${{ runner.os }}-
+      - name: Cache npm dependencies
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-npm
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
```
Comment (bpar476) [inline .github/workflows/mvn_test.yml]: You shouldn't need to cache the lock file as it's in VCS so will always be up to date. What needs to be cached is the `node_modules` directory.

Comment (bpar476) [reply .github/workflows/mvn_test.yml]: Whoops, I'm silly the yarn lock is just the hash file which is correct 👍 However, shouldn't the path be `node_modules`?

Comment (theghostwhoforks) [reply .github/workflows/mvn_test.yml]: https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

Yeah, it's just the hashfile!

I thought it should be `node_modules` too, but the doc asks to cache the `~/.npm` folder. Let me change this to the node_modules folder and see if it makes any difference in timing

Comment (theghostwhoforks) [reply .github/workflows/mvn_test.yml]: Screenshot 2020-03-16 at 12 45 27

Screenshot 2020-03-16 at 12 44 57

The cache size is the same (!) even while using the .npm or the node_modules folder

Comment (bpar476) [reply .github/workflows/mvn_test.yml]: That's very strange. I wonder if they have a weird config for storing node dependencies in the runner

Comment (bpar476) [reply .github/workflows/mvn_test.yml]: Looks like `~/.npm` is used to populate `node_modules` if it exists

Comment (theghostwhoforks) [reply .github/workflows/mvn_test.yml]: yeah. It stores the indexes and other cache data. Interesting.

**Pull request merged**